.clang-tidy

```diff
@@ -75,7 +75,37 @@ hicpp-*,
 -cppcoreguidelines-owning-memory,
 -cppcoreguidelines-init-variables,
 -hicpp-move-const-arg,
--hicpp-invalid-access-moved'
+-hicpp-invalid-access-moved,
+
+-clang-analyzer-core.uninitialized.UndefReturn,
+-clang-analyzer-optin.cplusplus.VirtualCall,
+
+-readability-qualified-auto,
+-readability-convert-member-functions-to-static,
+-readability-container-size-empty,
```

Comment:
I think this makes problem with `FunctionalInterface` and `NewType` but we can have exceptions for those